File: Creational-Patterns/C++/Builder/src/hardware/processor.cpp
#include "processor.hpp"


hardware_infos::Processor::Processor(   std::string processor_type,
                                        unsigned short int cache_number,
                                        unsigned short int cores,
                                        float clock )
{
    this->processor_type = processor_type;
    this->cache_number = cache_number;
    this->cores = cores;
    this->clock = clock;
}

std::string hardware_infos::Processor::get_processor_type()
{
    return this->processor_type;
}

unsigned short int hardware_infos::Processor::get_cache_number()
{
    return this->cache_number;
}

unsigned short int hardware_infos::Processor::get_cores()
{
    return this->cores;
}

float hardware_infos::Processor::get_clock()
{
    return this->clock;
}

File: Creational-Patterns/Python/Builder/hardware/processor.py
#coding: utf-8


class Processor(object):

    def __init__(self,processor_type: str, cores: int, clock: int, cache: int):
        self.__processor_type = processor_type
        self.__clock = clock
        self.__cache = cache
        self.__cores = cores

    def __str__(self):
        return_str = "== Processor ==\n"
        return_str += "Processor type : {}\n".format(self.__processor_type)
        return_str += "Processor clock: {}\n".format(self.__clock)
        return_str += "Processor cache: {}\n".format(self.__cache)
        return_str += "Processor cores: {}\n".format(self.__cores)
        return return_str
